winter_messaging_transactional/producer/publish_processor.py

```diff
@@ -37,22 +38,21 @@ def handle_interrupt_signal(signum, frame):
         signal.signal(signal.SIGINT, handle_interrupt_signal)
 
         is_error_occurred = False
-        while not cancel_token.wait(self._publish_interval):
-            outbox_messages = self._outbox_message_doa.select_unsent()
-            for outbox_message in outbox_messages:
-                exchange = self._topology_configurator.get_exchange_key(outbox_message.topic)
-                try:
-                    self._rabbitmq_client.publish(outbox_message, exchange)
-                    self._outbox_message_doa.mark_as_sent([outbox_message])
-                except MessageNotPublishedException:
-                    log.exception('Publishing processor error. Message not published: %s', outbox_message.message_id)
-                    is_error_occurred = True
+        while not cancel_token.wait(publish_interval):
+            with self._session.begin():
+                outbox_messages = self._outbox_message_doa.select_unsent()
+                for outbox_message in outbox_messages:
+                    exchange = self._topology_configurator.get_exchange_key(outbox_message.topic)
+                    try:
+                        self._rabbitmq_client.publish(outbox_message, exchange)
+                        self._outbox_message_doa.mark_as_sent([outbox_message])
+                    except MessageNotRoutedException:
+                        self._outbox_message_doa.mark_as_sent([outbox_message])
```

Comment:
if we catch it and do the same action like in normal sending case what the point raise it exception?